qiskit_dynamics/pulse/pulse_simulator.py

```diff
@@ -0,0 +1,330 @@
+
+        if solver_options is None:
+            solver_options = {}
+
+        if y0 is None:
+            y0 = Statevector(self._dressed_states[:, 0])
+
+        # to do: add handling of circuits
+        schedules = _to_schedule_list(run_input)
+
+        # get the acquires instructions and simulation times
+        t_span = []
+        measurement_subsystems_list = []
+        for schedule in schedules:
+            schedule_acquires = []
+            schedule_acquire_times = []
+            for start_time, inst in schedule.instructions:
+                if isinstance(inst, pulse.Acquire):
+                    schedule_acquires.append(inst)
+                    schedule_acquire_times.append(start_time)
+
+            # maybe need to validate more here
+            _validate_acquires(schedule_acquire_times, schedule_acquires)
+
+            t_span.append([0, schedule_acquire_times[0]])
+            measurement_subsystems_list.append([inst.channel.index for inst in schedule_acquires])
+            measurement_subsystems_list[-1].sort()
+
+
+        # Build and submit job
+        job_id = str(uuid.uuid4())
+        dynamics_job = DynamicsJob(
+            backend=self,
+            job_id=job_id,
+            fn=self._run,
+            fn_kwargs={
+                't_span': t_span, 'y0': y0, 'schedules': schedules,
+                'solver_options': solver_options,
+                'measurement_subsystems_list': measurement_subsystems_list,
+                'normalize_states': normalize_states,
+                'shots': shots
+            }
+        )
+        dynamics_job.submit()
+
+        return dynamics_job
+
+    def _run(
+        self,
+        job_id,
+        t_span,
+        y0,
+        schedules,
+        solver_options,
+        normalize_states,
+        measurement_subsystems_list,
+        shots
+    ) -> Result:
+        """Not sure here what the right delineation of arguments is to put in _run.
+        This feels somewhat hacky/arbitrary.
+        """
+        start = time.time()
+
+        # map measurement subsystems from labels to correct index
+        if self.subsystem_labels:
+            new_measurement_subsystems_list = []
+            for measurement_subsystems in measurement_subsystems_list:
+                new_measurement_subsystems = []
+                for subsystem in measurement_subsystems:
+                    if subsystem in self.subsystem_labels:
+                        new_measurement_subsystems.append(self.subsystem_labels.index(subsystem))
+                    else:
+                        raise QiskitError(f"Attempted to measure subsystem {subsystem}, but it is not in subsystem_list.")
+                new_measurement_subsystems_list.append(new_measurement_subsystems)
+
+            measurement_subsystems_list = new_measurement_subsystems_list
+
+        solver_results = self.solver.solve(t_span=t_span, y0=y0, signals=schedules, **solver_options)
+
+        # construct counts for each experiment
```

Comment:
I think you should wrap up measurement sampling into a separate function, so that you could pass in custom measurement functions as an option (and have some pre-defined ones you can choose from with string names)